### PR TITLE
[ENH] Hierarchical scitype support for polars  

### PR DESCRIPTION
**Pull request by pranavvp16**
This PR is complete with checks and conversion functions for Hierarchical scitype for polars, with scitype name `polars_hierarchical`. PR #6485 and #6552 needs to be merged before this

### CODE REVIEW

**Comment by pranavvp16** [comment]
@fkiraly This PR is also ready to merge as #6552 is merged now. I have also updated this branch with

File: sktime/datatypes/_panel/_check.py [hunk 1]
```diff
@@ -545,12 +545,14 @@ def check_dask_panel(obj, return_metadata=False, var_name="obj"):
 
     from sktime.datatypes._adapter.polars import check_polars_frame, get_mi_cols
 
-    def check_polars_panel(obj, return_metadata=False, var_name="obj", panel=True):
+    def check_polars_panel(
```
Comment (fkiraly) [inline sktime/datatypes/_panel/_check.py:548]: why are you changing this? The arg `panel` should actually be removed.

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks for the reminder!

* I think the "examples" part whould reintroduce the failure of `sktime` on `polars` described here - https://github.com/sktime/sktime/issues/7056 - can you kindly fix that?
* the checker should have only the same arguments every other checker has. Internally, of course, it can call the more general checker.

**Review by fkiraly — CHANGES_REQUESTED** [review]
@pranavvp16, I have now updated `main` with the handling of the soft dependencies in examples, and also adapted the `_examples` file - could you kindly have a look?

**Comment by pranavvp16** [comment]
Meanwhile I was thinking for a smaller fix the new `datatypes` example refactor solves this issue more efficiently. I have addressed the remaining comments in this PR, can you kindly review it @fkiraly .

**Review by fkiraly — APPROVED** [review]
Looks good to me now, if you are also happy with it.

**Review by fkiraly — APPROVED** [review]

**Pull request merged**